File: client/src/types/movie.ts
export interface MovieDto {
  id: string;
  title: string;
  year?: number;
  posterUrl?: string;
  rating?: number;
  genres: string[];
}

export interface WatchlistDto {
  id: string;
  name: string;
  lastRefreshedAt?: string;
  movies: MovieDto[];
}

export interface MovieFilterRequest {
  query?: string;
  yearFrom?: number;
  yearTo?: number;
  minRating?: number;
  genres?: string[];
}

File: ImdbWatchlists/Models/Movie.cs
namespace ImdbWatchlists.Models;

public record Movie
{
    public required string Id { get; init; }

    public required string Title { get; init; }

    public int? Year { get; init; }

    public string? PosterUrl { get; init; }

    public double? Rating { get; init; }

    public IReadOnlyCollection<string> Genres { get; init; } = [];
}
